README.md
# VAGABOND HIATUS CHART

<p align="center">
  	<img src="docs/images/vagabond_banner.jpg" />
</p>

## :information_source: About

A simple way to visualize the actual state of Vagabond by Inoue Takehiko with charts.

## :rocket: Check the site

[Vagabond Hiatus Chart](https://vagabond-hiatus-chart.vercel.app/)

## :open_file_folder: Structure

| Codebase          | Description      | Status                    | Docs                     |
| :---------------- | :--------------- | :------------------------ | :----------------------- |
| [moon](apps/moon) | Create React App | :x: deprecated            | [Readme](./docs/MOON.md) |
| [sun](apps/sun)   | Next.js          | :white_check_mark: active | [Readme](./docs/SUN.md)  |

## :clap: Special Thanks

- Kawasaki#3316 for providing almost the entire [dataset](https://docs.google.com/spreadsheets/d/1fw7G9I2zPtAfSh0NUl-4m7G5wsXe5PIcMFRtd03jVz0/). without their contribution, this project would be very inaccurate (the current data is collected from the morning weekly website).

- [rentzhx3](https://github.com/rentzhx3/) for inspiring this project with their work based on Hunter X Hunter.

## :page_with_curl: License

[GPL-3.0 License](LICENSE)

## :scroll: Changelog

[Last update: 26/12/23](./docs/CHANGELOG.md)


docs/CHANGELOG.md
# :clock12: Changelog

## :date: v1.0.6

<details><summary>Dec 26, 2023</summary>

**project**

- add: use workspace from pnpm
- update: `.devcontainer`.
- update: labeler.
- update: pnpm version.
- update: Node.js version development.

**moon**

- fix: format files.
- remove(deps): prettier.

**sun**

- update: db `new morning issue` at `#53`.
- update(deps): to latest.
- remove(deps): prettier.

</details>

## :date: v1.0.5

<details><summary>Oct 03, 2023</summary>

**project**

- add: `.devcontainer`.
- update: labeler.

**sun**

- update: db `new morning issue` at `#45`.
- update(deps): to latest.

</details>

## :date: v1.0.4

<details><summary>Aug 22, 2023</summary>

**project**

- add: `Tailwind CSS` extension.
- update: README.
- update: `actions/labeler` config.
- remove: `styled-components` extension.

**sun**

- add: `Tailwind CSS`.
- add: dark and light theme support.
- update: use the new `app` router.
- update: db `new morning issue` at `#39`.
- update: `.eslintrc` config.
- update(deps): to latest.
- remove: `styled-components`.

</details>

## :date: v1.0.3

<details><summary>Jul 15, 2023</summary>

**project**

- update: README.
- update(.vscode): `bracketPairs` change **true**, enable `linkedEditing`.

**moon**

- remove keys `keywords`, `author` and `license` from package.json.

**sun**

- add: new prefix `$` for styled-components `props`.
- update: db `new morning issue` at `#34`.
- update: engine `node` v18 and `pnpm` v8.
- update(deps): to latest.
- remove: keys `name`,`author`, `description`, `keywords`, `author` and `license` from package.json.
- remove: don't commit `next-env.d.ts` file.
- remove: `.eslintignore` file, instead use `ignorePatterns` in **.eslintrc.json** file.

</details>

## :date: v1.0.2

<details><summary>Sep 30, 2022</summary>

**project**

- add Sun and Moon docs.
- add .editorconfig.
- remove node ci
- update README.

**moon**

- add redirect to vercel site

**sun**

- Remove ApexChart Heatmap.
- Add pnpm as package manager.
- Add @nivo/heatmap.
- Add heatmap legends.
- Update db.

</details>

## :date: v1.0.1

<details><summary>Aug 31, 2022</summary>

**project**

- fix github workflows (labeler, node).
- setup settings from vscode.
- add vercelignore.
- add changelog in readme(root).

**moon**

- update deps (prettier, react-router-dom)
- update db weekly issues (current at #40) from moon.
- build the website and publish on gh-pages branch.
- remove unused nivo script.
- remove theme provider from styled components.
- remove duplicated styles.
- remove glitch effect in 404 page aka nomatch.
- fixed readme, prettier, prettierignore.
- add disabled property in heatmap buttons.
- add prettier --cache flag in package.json script.

**sun**

- fix and use my own apexchart wrapper.
- add gradients in svg vectors.
</details>

## :date: v1.0.0

<details><summary>Jun 16, 2022</summary>

**project**

- CRA in [moon](moon) codebase deprecated, starting on NEXT JS [sun](sun) codebase [`e0e417a`](https://github.com/falsepopsky/vagabond-hiatus-chart/commit/e0e417a73a2cc0aeeddab368b1cc28ab047e2d81)
</details>


src/data/sample.js
const Data = [
  {
    country: 'AD',
    'hot dog': 50,
    'hot dogColor': 'hsl(212, 74%, 64%)',
    burger: 35,
    burgerColor: 'hsl(118, 74%, 64%)',
    sandwich: 10,
    sandwichColor: 'hsl(38, 70%, 50%)',
    kebab: 24,
    kebabColor: 'hsl(315, 70%, 50%)',
    fries: 20,
    friesColor: 'hsl(242, 70%, 50%)',
    donut: 29,
    donutColor: 'hsl(253, 70%, 50%)',
    junk: 47,
    junkColor: 'hsl(349, 70%, 50%)',
    sushi: 77,
    sushiColor: 'hsl(315, 70%, 50%)',
    ramen: 12,
    ramenColor: 'hsl(2, 70%, 50%)',
    curry: 64,
    curryColor: 'hsl(3, 70%, 50%)',
    udon: 24,
    udonColor: 'hsl(260, 70%, 50%)',
  },
  {
    country: 'AE',
    'hot dog': 86,
    'hot dogColor': 'hsl(131, 70%, 50%)',
    burger: 59,
    burgerColor: 'hsl(21, 70%, 50%)',
    sandwich: 93,
    sandwichColor: 'hsl(21, 70%, 50%)',
    kebab: 96,
    kebabColor: 'hsl(339, 70%, 50%)',
    fries: 0,
    friesColor: 'hsl(325, 70%, 50%)',
    donut: 4,
    donutColor: 'hsl(205, 70%, 50%)',
    junk: 19,
    junkColor: 'hsl(143, 70%, 50%)',
    sushi: 12,
    sushiColor: 'hsl(118, 70%, 50%)',
    ramen: 27,
    ramenColor: 'hsl(93, 70%, 50%)',
    curry: 44,
    curryColor: 'hsl(349, 70%, 50%)',
    udon: 38,
    udonColor: 'hsl(295, 70%, 50%)',
  },
  {
    country: 'AF',
    'hot dog': 8,
    'hot dogColor': 'hsl(9, 70%, 50%)',
    burger: 59,
    burgerColor: 'hsl(49, 70%, 50%)',
    sandwich: 49,
    sandwichColor: 'hsl(64, 70%, 50%)',
    kebab: 74,
    kebabColor: 'hsl(137, 70%, 50%)',
    fries: 21,
    friesColor: 'hsl(302, 70%, 50%)',
    donut: 86,
    donutColor: 'hsl(286, 70%, 50%)',
    junk: 52,
    junkColor: 'hsl(179, 70%, 50%)',
    sushi: 28,
    sushiColor: 'hsl(53, 70%, 50%)',
    ramen: 31,
    ramenColor: 'hsl(326, 70%, 50%)',
    curry: 53,
    curryColor: 'hsl(354, 70%, 50%)',
    udon: 36,
    udonColor: 'hsl(35, 70%, 50%)',
  },
  {
    country: 'AG',
    'hot dog': 37,
    'hot dogColor': 'hsl(215, 70%, 50%)',
    burger: 79,
    burgerColor: 'hsl(124, 70%, 50%)',
    sandwich: 67,
    sandwichColor: 'hsl(267, 70%, 50%)',
    kebab: 85,
    kebabColor: 'hsl(104, 70%, 50%)',
    fries: 31,
    friesColor: 'hsl(341, 70%, 50%)',
    donut: 95,
    donutColor: 'hsl(191, 70%, 50%)',
    junk: 31,
    junkColor: 'hsl(138, 70%, 50%)',
    sushi: 62,
    sushiColor: 'hsl(136, 70%, 50%)',
    ramen: 97,
    ramenColor: 'hsl(262, 70%, 50%)',
    curry: 12,
    curryColor: 'hsl(237, 70%, 50%)',
    udon: 82,
    udonColor: 'hsl(221, 70%, 50%)',
  },
  {
    country: 'AI',
    'hot dog': 31,
    'hot dogColor': 'hsl(159, 70%, 50%)',
    burger: 54,
    burgerColor: 'hsl(103, 70%, 50%)',
    sandwich: 0,
    sandwichColor: 'hsl(191, 70%, 50%)',
    kebab: 55,
    kebabColor: 'hsl(176, 70%, 50%)',
    fries: 19,
    friesColor: 'hsl(225, 70%, 50%)',
    donut: 78,
    donutColor: 'hsl(52, 70%, 50%)',
    junk: 52,
    junkColor: 'hsl(110, 70%, 50%)',
    sushi: 12,
    sushiColor: 'hsl(297, 70%, 50%)',
    ramen: 94,
    ramenColor: 'hsl(241, 70%, 50%)',
    curry: 12,
    curryColor: 'hsl(352, 70%, 50%)',
    udon: 66,
    udonColor: 'hsl(22, 70%, 50%)',
  },
  {
    country: 'AL',
    'hot dog': 5,
    'hot dogColor': 'hsl(283, 70%, 50%)',
    burger: 52,
    burgerColor: 'hsl(340, 70%, 50%)',
    sandwich: 94,
    sandwichColor: 'hsl(341, 70%, 50%)',
    kebab: 54,
    kebabColor: 'hsl(256, 70%, 50%)',
    fries: 18,
    friesColor: 'hsl(180, 70%, 50%)',
    donut: 42,
    donutColor: 'hsl(342, 70%, 50%)',
    junk: 12,
    junkColor: 'hsl(94, 70%, 50%)',
    sushi: 24,
    sushiColor: 'hsl(282, 70%, 50%)',
    ramen: 18,
    ramenColor: 'hsl(53, 70%, 50%)',
    curry: 29,
    curryColor: 'hsl(332, 70%, 50%)',
    udon: 52,
    udonColor: 'hsl(151, 70%, 50%)',
  },
  {
    country: 'AM',
    'hot dog': 34,
    'hot dogColor': 'hsl(205, 70%, 50%)',
    burger: 15,
    burgerColor: 'hsl(307, 70%, 50%)',
    sandwich: 69,
    sandwichColor: 'hsl(272, 70%, 50%)',
    kebab: 65,
    kebabColor: 'hsl(69, 70%, 50%)',
    fries: 66,
    friesColor: 'hsl(271, 70%, 50%)',
    donut: 37,
    donutColor: 'hsl(195, 70%, 50%)',
    junk: 82,
    junkColor: 'hsl(168, 70%, 50%)',
    sushi: 29,
    sushiColor: 'hsl(50, 70%, 50%)',
    ramen: 66,
    ramenColor: 'hsl(335, 70%, 50%)',
    curry: 14,
    curryColor: 'hsl(214, 70%, 50%)',
    udon: 13,
    udonColor: 'hsl(49, 70%, 50%)',
  },
  {
    country: 'AO',
    'hot dog': 97,
    'hot dogColor': 'hsl(95, 70%, 50%)',
    burger: 61,
    burgerColor: 'hsl(128, 70%, 50%)',
    sandwich: 31,
    sandwichColor: 'hsl(13, 70%, 50%)',
    kebab: 60,
    kebabColor: 'hsl(193, 70%, 50%)',
    fries: 16,
    friesColor: 'hsl(292, 70%, 50%)',
    donut: 0,
    donutColor: 'hsl(9, 70%, 50%)',
    junk: 29,
    junkColor: 'hsl(295, 70%, 50%)',
    sushi: 56,
    sushiColor: 'hsl(115, 70%, 50%)',
    ramen: 83,
    ramenColor: 'hsl(225, 70%, 50%)',
    curry: 83,
    curryColor: 'hsl(155, 70%, 50%)',
    udon: 84,
    udonColor: 'hsl(11, 70%, 50%)',
  },
  {
    country: 'AQ',
    'hot dog': 72,
    'hot dogColor': 'hsl(24, 70%, 50%)',
    burger: 20,
    burgerColor: 'hsl(54, 70%, 50%)',
    sandwich: 42,
    sandwichColor: 'hsl(304, 70%, 50%)',
    kebab: 82,
    kebabColor: 'hsl(214, 70%, 50%)',
    fries: 39,
    friesColor: 'hsl(5, 70%, 50%)',
    donut: 86,
    donutColor: 'hsl(25, 70%, 50%)',
    junk: 30,
    junkColor: 'hsl(251, 70%, 50%)',
    sushi: 24,
    sushiColor: 'hsl(24, 70%, 50%)',
    ramen: 81,
    ramenColor: 'hsl(108, 70%, 50%)',
    curry: 83,
    curryColor: 'hsl(146, 70%, 50%)',
    udon: 24,
    udonColor: 'hsl(64, 70%, 50%)',
  },
];

export default Data;
